akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -1609,7 +1609,23 @@ final class Flow[-In, +Out, +Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends
     new Flow(delegate.idleTimeout(timeout))
 
   /**
-   * Injects additional elements if the upstream does not emit for a configured amount of time. In other words, this
+    * If the time between the emission of an element and the following downstream demand exceeds the provided timeout,
+    * the stream is failed with a [[java.util.concurrent.TimeoutException]]. If the producer doesn't emmit
+    * any element, the timeout is still valid and will fail the stream.
```

Comment:
Maybe add a comment that for efficiency the timeout is checked only periodically, so the resolution of the check is one period.